crates/bevy_ui/src/layout/ui_surface.rs

```diff
@@ -0,0 +1,607 @@
+        let camera_entity = Entity::from_raw(0);
+        let root_node_entity = Entity::from_raw(1);
+        let style = Style::default();
+
+        ui_surface.upsert_node(root_node_entity, &style, &DUMMY_LAYOUT_CONTEXT);
+
+        // assign root node to camera
+        ui_surface.set_camera_children(camera_entity, [root_node_entity].into_iter());
+
+        assert!(ui_surface
+            .camera_entity_to_taffy
+            .contains_key(&camera_entity));
+        assert!(ui_surface
+            .camera_entity_to_taffy
+            .get(&camera_entity)
+            .unwrap()
+            .contains_key(&root_node_entity));
+        assert!(ui_surface.camera_roots.contains_key(&camera_entity));
+        let root_node_pair = ui_surface
+            .get_root_node_pair_exact(root_node_entity, camera_entity)
+            .expect("expected root node pair");
+        assert!(ui_surface
+            .camera_roots
+            .get(&camera_entity)
+            .unwrap()
+            .contains(root_node_pair));
+
+        ui_surface.remove_camera_entities([camera_entity]);
+
+        // should not affect `entity_to_taffy`
+        assert!(ui_surface.entity_to_taffy.contains_key(&root_node_entity));
+
+        // `camera_roots` and `camera_entity_to_taffy` should no longer contain entries for `camera_entity`
+        assert!(!ui_surface
+            .camera_entity_to_taffy
+            .contains_key(&camera_entity));
+        return; // TODO: can't pass the test if we continue - not implemented
+        assert!(!ui_surface.camera_roots.contains_key(&camera_entity));
+
+        // root node pair should be removed
+        let root_node_pair = ui_surface.get_root_node_pair_exact(root_node_entity, camera_entity);
+        assert_eq!(root_node_pair, None);
+    }
+
+    #[test]
+    fn test_remove_entities() {
+        let mut ui_surface = UiSurface::default();
+        let camera_entity = Entity::from_raw(0);
+        let root_node_entity = Entity::from_raw(1);
+        let style = Style::default();
+
+        ui_surface.upsert_node(root_node_entity, &style, &DUMMY_LAYOUT_CONTEXT);
+
+        ui_surface.set_camera_children(camera_entity, [root_node_entity].into_iter());
+
+        assert!(ui_surface.entity_to_taffy.contains_key(&root_node_entity));
+        assert!(ui_surface
+            .camera_entity_to_taffy
+            .get(&camera_entity)
+            .unwrap()
+            .contains_key(&root_node_entity));
+        let root_node_pair = ui_surface
+            .get_root_node_pair_exact(root_node_entity, camera_entity)
+            .unwrap();
+        assert!(ui_surface
+            .camera_roots
+            .get(&camera_entity)
+            .unwrap()
+            .contains(root_node_pair));
+
+        ui_surface.remove_entities([root_node_entity]);
+        assert!(!ui_surface.entity_to_taffy.contains_key(&root_node_entity));
+
+        return; // TODO: can't pass the test if we continue - not implemented
+        assert!(!ui_surface
```

Comment:
What's the preferred way to handle this:

a. `#[allow(unreachable_code)]`
b. comment out the unreachable code

(return removed in https://github.com/bevyengine/bevy/pull/12804 where the full tests can be completed)

Reply:
Went with `#[allow(unreachable_code)]` for now

Reply:
So this is just here because the pr that this pr is blocking will solve it?

Reply:
Correct, iirc it was a negative/regression test that I wasn't quite sure what do with when I was splitting my changes up. It's acknowledging that we are failing to fully clean up some of the state. Figured leaving it in for documentation purposes was worth it, if the other PR solving it never makes it in or becomes delayed significantly.